package/MDAnalysis/analysis/waterdynamics.py

```diff
@@ -1194,99 +1192,88 @@ class SurvivalProbability(object):
       Universe object
     selection : str
       Selection string; any selection is allowed. With this selection you
-      define the region/zone where to analyze, e.g.: "selection_a" and "zone"
-      (see `SP-examples`_ )
-    t0 : int
-      frame  where analysis begins
-    tf : int
-      frame where analysis ends
-    dtmax : int
-      Maximum dt size, `dtmax` < `tf` or it will crash.
-
+      define the region/zone where to analyze, e.g.: "resname SOL and around 5 (resname LIPID)"
+      and "resname ION and around 10 (resid 20)" (see `SP-examples`_ )
+    verbose : Boolean
+      If True, prints progress and comments to the console.
 
     .. versionadded:: 0.11.0
 
     """
 
-    def __init__(self, universe, selection, t0, tf, dtmax):
+    def __init__(self, universe, selection, verbose=False):
         self.universe = universe
         self.selection = selection
-        self.t0 = t0
```

Comment:
So we can't just stop allowing `t0` and `tf` here, because it would break people's existing code.  What we need to do instead is to allow it, but raise warnings if it's used, so something like:


```python
def __init__(...., t0=None, tf=None):
    if not t0 is None:
        self.start = t0
        warnings.warn("t0 is deprecated, use run(start=t0) instead", category=DeprecationWarning)

Reply:
Updated, thanks.